### PR TITLE
Kevin.cranmer/api 8161 invalid part logging

### PR DESCRIPTION
**Pull request by KevinCranmer**
## Description of change

This PR adds logging and slack notifying for submissions submitted with invalid attachment names. The information is for internal research at the moment. We plan to use this information to decide if errors should be reported to users if they submit invalid attachment names.

## Original issue(s)
https://vajira.max.gov/browse/API-8161

## Things to know about this PR

Additions to the settings.yml file were made just to give some default values for other devs. There is another PR for getting the necessary Setting parameter added in here: https://github.com/department-of-veterans-affairs/devops/pull/9758

Automated postman tests were ran for various names of attachments. Rspec tests added for the slack notification job.

### CODE REVIEW

**Review by mviterna — APPROVED** [review]

File: modules/vba_documents/app/workers/vba_documents/slack_notifier.rb [hunk 1]
```diff
@@ -66,10 +69,32 @@ def long_flyers_alert
...
+        end
+      end
+      resp&.success?
+    end
+
     def send_to_slack(text)
       Faraday.post(@slack_url, "{\"text\": \"#{text}\"}", 'Content-Type' => 'application/json')
     end
 
+    def send_to_internal_slack(text)
```
Comment (cshupp1) [inline modules/vba_documents/app/workers/vba_documents/slack_notifier.rb:94]: Can we refactor send_to_slack such that it is:
def send_to_slack(text, url) ?

File: modules/vba_documents/app/workers/vba_documents/slack_notifier.rb [hunk 2]
```diff
@@ -27,6 +29,7 @@ def perform
 
     def fetch_settings
       @slack_url = Settings.vba_documents.slack.notification_url
+      @internal_slack_url = Settings.vba_documents.slack.internal_notification_url
```
Comment (cshupp1) [inline modules/vba_documents/app/workers/vba_documents/slack_notifier.rb:32]: Well, you have shined a light on my bad name of @slack_url.  I like your name invalid_parts_alerted.

Perhaps a rename:
@slack_url => @default_alert_url
@internal_slack_url => @invalid_parts_alert_url

File: modules/vba_documents/spec/jobs/slack_notifier_spec.rb [hunk 3]
```diff
@@ -146,4 +150,24 @@
     u = VBADocuments::UploadSubmission.find_by(guid: guid)
     expect(last_notified).to be < u.metadata['last_slack_notification'].to_i
   end
+
+  context 'invalid parts' do
```
Comment (cshupp1) [inline modules/vba_documents/spec/jobs/slack_notifier_spec.rb:154]: Nice tests

File: modules/vba_documents/spec/jobs/slack_notifier_spec.rb [hunk 4]
```diff
@@ -12,6 +12,7 @@
       update_stalled_notification_in_minutes: 180,
       daily_notification_hour: 7,
       notification_url: '',
+      internal_notification_url: '',
```
Comment (cshupp1) [inline modules/vba_documents/spec/jobs/slack_notifier_spec.rb:15]: Same naming comment here.

File: config/settings.yml [hunk 5]
```diff
@@ -464,6 +464,16 @@ vba_documents:
     topic_arns:
       - vetsgov-arn
       - vagov-arn
+  slack:
+    enabled: false
+    notification_url:
+    internal_notification_url:
```
Comment (cshupp1) [inline config/settings.yml:470]: See the naming comment below.  I don't have a handle on the credstash changes yet, but I assume changing line 468 is doable too.  Is it?

**Review by cshupp1 — APPROVED** [review]

**Pull request merged**